src/compiler/checker.ts

```diff
@@ -11252,6 +11251,27 @@ namespace ts {
                         }
                         return;
                     }
+                    else if (target.flags & TypeFlags.IndexedAccess) {

```

Comment:
The odd location of this branch within `inferTypes` is to work around #20124 - if I placed it below the other `IndexedAccess` case (as I'd prefer to do), it would never be executed because of that bug.